jumpscale/core/base/fields.py

```diff
@@ -105,5 +106,163 @@ def validate(self, value):
             self.field.validate(item)
 
 
+class Email(Field):
+    def __init__(self, default="", **kwargs):
+        super().__init__(default, **kwargs)
+        self.regex = r"^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$"
+
+    def validate(self, value):
+        """Check whether provided value is a valid email representation
+        Args:
+            value (str)
+        Returns:
+            Boolean expresion"""
+        super().validate(value)
+        return re.search(self.regex, value) is not None
+
+
+class Path(Field):
+    def __init__(self, default="", **kwargs):
+        super().__init__(default, **kwargs)
+        self.regex = r"^(/[^/ ]*)+/?$"
+
+    def validate(self, value):
+        """Check whether provided value is a valid path representation
+        Args:
+            value (str)
+        Returns:
+            Boolean expresion"""
+        super().validate(value)
+        return re.search(self.regex, value) is not None
+
+
+class URL(Field):
+    def __init__(self, default="", **kwargs):
+        super().__init__(default, **kwargs)
+        self.regex = r"^(https?|ftp)://[^\s/$.?#].[^\s]*$"
+
+    def validate(self, value):
+        """Check whether provided value is a valid URL representation
+        Args:
+            value (str)
+        Returbs:
+            Boolean expresion"""
+        super().validate(value)
+        return re.search(self.regex, value) is not None
```

Comment:
lets use urllib.parse to check here against emptry schema, .. etc

Reply:
I have used ```urllib.parse``` to validate if the URL has a schema  and domain so it is valid URL